### ISSUE TITLE
[Python] Probabilities flag is not working

### ISSUE THREAD

**Issue by montenegrodr** [JohnLangford/vowpal_wabbit#1113]
This snippet does not return the expected result:

```
from vowpalwabbit.sklearn_vw import VW

X = ['1 | feature1:2.5',
     '2 | feature1:0.11 feature2:-0.0741',
     '3 | feature3:2.33 feature4:0.8 feature5:-3.1',
     '1 | feature2:-0.028 feature1:4.43',
     '2 | feature5:1.532 feature6:-3.2']

model = VW(
    convert_to_vw=False, loss_function='logistic', oaa=3, probabilities=True
)
model.fit(X)
print model.predict([' | feature1:0.11 feature2:-0.0741'])
```

That results in something like:
**`[ 28036672.]`**

It shouldn't be probabilities like:
**`[0.3, 0.6, 0.1]`**

Thanks.

**Comment by JohnLangford** [comment]
@gramhagen @hal3 I suspect this is a type conversion error.  This should be much more easily avoided now that the prediction type is explicit in the top level learner.

**Comment by gramhagen** [comment]
there's a couple things going on. right now the file exposing methods to python (pylibvw.cc) doesn't have support for arrays of probabilities. I'm not super familiar with how to convert c++ arrays to python objects, so my initial attempt failed.

```
v_array<float> ex_get_multiclass_probabilities(example_ptr ec) { return ec->pred.scalars; }
```

I'm sure there's probably a way around that, but it will take a little manipulating. Accessing the prediction still seems to give the large integer value you noticed, so there's still something funky with how it's accessing the label here:

```
uint32_t ex_get_multiclass_label(example_ptr ec) { return ec->l.multi.label; }
```

I'm a little out of my depth on the c++ side, though. so I'm not completely sure what we should be accessing if not l.multi.label.

**Comment by JohnLangford** [comment]
(I'm similarly ill-informed about the python side.)

To return a vector of probabilities, you need to add --probabilities to the
commandline---otherwise if defaults to returning a single value.  Also
v_array<float> is a type that python is very unlikely to recognize.

I'm a little bit confused about why you mention the label when you want to
predict.  To get the prediction, you want something like ec->pred.scalar
Could that be the issue?

-John

**Comment by gramhagen** [comment]
ok, np I was a bit confused about the labels (not realizing that they have to be encoded as ints)

I can make a new method that passes a list back to python to handle providing multiple probabilities stored in ec->pred.scalars. And I'll update the predict method to catch the --probabilities case and call the new method.

As for the sklearn interface, it would be better to wrap this under sklearn.multiclass.OneVsRestClassifier, but I'll try to get the first part out sooner so it can be used without sklearn if necessary.

**Comment by JohnLangford** [comment]
Thanks Scott.

-John

**Comment by JohnLangford** [comment]
This should be fixed now.  Thanks @gramhagen
